crates/bifrost/src/providers/replicated_loglet/sequencer/node.rs

```diff
@@ -0,0 +1,269 @@
+use std::{collections::BTreeMap, sync::Arc};
+
+use super::{
+    worker::{NodeWorker, NodeWorkerHandle, Payload, SendPermit},
+    Error, Node, NodeClient, SequencerGlobalState,
+};
+use restate_core::TaskCenter;
+use restate_types::{
+    logs::{LogletOffset, SequenceNumber},
+    net::log_server::LogletInfo,
+    PlainNodeId,
+};
+
+#[derive(derive_more::Debug, derive_more::Deref, derive_more::DerefMut)]
+#[debug("{:?}", inner.keys().collect::<Vec<&PlainNodeId>>())]
+/// Nodes represents the entire cluster of nodes available. This can be shared with
+/// multiple Sequencers
+pub struct Nodes<C> {
+    #[deref]
+    #[deref_mut]
+    inner: BTreeMap<PlainNodeId, Node<C>>,
+}
+
+impl<C, I> From<I> for Nodes<C>
+where
+    I: IntoIterator<Item = (PlainNodeId, C)>,
+{
+    fn from(value: I) -> Self {
+        let inner: BTreeMap<PlainNodeId, Node<C>> = value
+            .into_iter()
+            .map(|(key, client)| (key, Node::new(client)))
+            .collect();
+
+        Self { inner }
+    }
+}
+
+#[derive(Debug, Clone)]
+pub struct LogletStatus {
+    pub sealed: bool,
+    pub local_tail: LogletOffset,
+    pub inflight_tail: LogletOffset,
+    pub trim_point: LogletOffset,
+}
+
+impl Default for LogletStatus {
+    fn default() -> Self {
+        Self {
+            sealed: false,
+            local_tail: LogletOffset::INVALID,
+            inflight_tail: LogletOffset::INVALID,
+            trim_point: LogletOffset::INVALID,
+        }
+    }
+}
+
+impl LogletStatus {
+    pub fn update(&mut self, info: &LogletInfo) {
+        self.sealed = info.sealed;
+        self.local_tail = info.local_tail;
+        self.trim_point = info.trim_point;
+    }
+}
+
+#[derive(Debug)]
+pub struct LogletNode {
+    pub loglet: LogletStatus,
+    pub worker: NodeWorkerHandle,
+}
+
+/// NodeSet is a subset of Nodes that is maintained internally with each sequencer
+#[derive(derive_more::Deref, derive_more::DerefMut, Debug)]
+pub struct NodeSet {
```

Comment:
That's going to be very confusing name given that we have a different meaning for NodeSet. Can we call this `ActiveLogServers`?